includes/class-wc-taxjar-integration.php

```diff
@@ -839,21 +839,42 @@ public function smartcalcs_cache_request( $json ) {
 		}
 
 		/**
-		 * Calculate tax / totals using TaxJar at checkout
-		 *
-		 * @return void
+         * Determines whether TaxJar should calculate tax on the cart
+         *
+		 * @param WC_Cart $wc_cart_object
+		 * @return bool - whether or not TaxJar should calculate tax
 		 */
-		public function calculate_totals( $wc_cart_object ) {
+		public function should_calculate_cart_tax( $wc_cart_object ) {
+		    $should_calculate = true;
+
 			// If outside of cart and checkout page or within mini-cart, skip calculations
 			if ( ( ! is_cart() && ! is_checkout() ) || ( is_cart() && is_ajax() ) ) {
-				return;
+				$should_calculate = false;
 			}
 
 			// prevent unnecessary calls to API during add to cart process
 			if ( doing_action( 'woocommerce_add_to_cart' ) ) {
-				return;
+				$should_calculate = false;
 			}
 
+			if ( floatval( $wc_cart_object->get_total( null ) ) === 0.0 ) {
+				$should_calculate = false;
+            }
+
+		    return apply_filters( 'taxjar_should_calculate_cart_tax', $should_calculate );
```

Comment:
What do you think about adding `$wc_cart_object` as an additional argument?

```
return apply_filters( 'taxjar_should_calculate_cart_tax', $should_calculate, $wc_cart_object );
```

That would allow someone to conditionally calculate tax based on the cart object.

Reply:
Thanks for catching this, yep it should definitely have the cart object in that filter. We will make that change before it gets released.